src/java.desktop/macosx/classes/sun/lwawt/LWWindowPeer.java

```diff
@@ -196,7 +196,7 @@ public LWWindowPeer(Window target, PlatformComponent platformComponent,
         }
 
         platformWindow.initialize(target, this, ownerDelegate);
-
+        platformWindow.setOpaque(!isTranslucent());
```

Comment:
!isTranslucent() uses the opaque flag which will be initialised in the code below(see setOpaque(getTarget().isOpaque());)

Reply:
Yes, and we need this call here to initialise content view with correct default value
CPlatformWindow.java:931 contentView.setWindowLayerOpaque(isOpaque);

Reply:
I need to look at it closly, I do not understand why we made the native layer opaque=NO by default while the java code uses opaque=true by default. And then we reset the native code from NO to yes by the code above. Why we cannot make both native/java use true/YES and then necessary it will be reset to false/NO by the existing code(via LWWindowPeer.setOpaque() and LWWindowPeer.updateOpaque?)
BTW Probably implementation of LWWindowPeer.setTextured() should call the updateOpaque() as well.

Reply:
Yes, sounds reasonable. Moreover, It was my initial attempt but I faced with some regressions in SwingSet2 demo. In some cases background of tree nodes  become black and this artifact was reproducible only if some other tabs were clicked before.  It was hot time for jdk17 so I decided not to fight with this regression. Now we have more time, so I can spend more time on this.

Reply:
Looks like making MTLLayer opaque by default require much more time that I expected. I tried different things to resolve artefacts that I mentioned before but nothing seems to work. So, I suggest to stick with my latest approach. It passes all the tests and don't break SwingSet2.

> BTW Probably implementation of LWWindowPeer.setTextured() should call the updateOpaque() as well.

I'll add this change

Reply:
How to reproduce that artifact?

Reply:
1. Run SwingSet2
2. Click on all tabs starting from the second one 
3. On the tab with swing tree control try to expand some tree nodes -> you'll see black rectangles while performing  the clicks

It may not be reproduced on a first attempt. On my machine it's reproducible with probability at least 50%

Reply:
Thank you, looking into it.

Reply:
I check the steps below and was able to reproduce the same artifacts(black rectangles) in the tree tab with the current version of the fix. And I do not see a big difference between setting the layer to YES at the beginning or a little bit later, at the end, it will be opaque and artifacts occur.

1. Run SwingSet2
2. Click on all tabs starting from the second one, delay on each tab for the second or so
3. On the tab with swing tree control try to expand some tree nodes -> you'll see black rectangles while performing the clicks
4. Repeat from step 2 if no artifacts are visible

Reply:
Interesting, now I'm also able to reproduce this problem with the latest version of the fix. Actually, looks like the problem exists even without the fix. We get invalid background with the same scenario. It's not so visible because of default L&F (light). Nevertheless, the color of the background is not exact the same. If we provide apple.awt.application.appearance=NSAppearanceNameDarkAqua  VM option, the problem became visible with metal and not reproducible with OGL.

Reply:
I've filed a separate CR for this problem JDK-8272392

Reply:
Then you can roll back this change to the default opaque layer?

Reply:
Did you mean to make MTLLayer opaque by default? Yes, I did it in the latest version.

Reply:
Then why do you need " platformWindow.setOpaque(!isTranslucent());" above?

Reply:
Ah, I see. Updated.